File: Java/hellointerview/01-two-pointers/max_area.java
public class Solution {
  public Integer max_area(int[] heights) {
    var left = 0;
    var right = heights.length - 1;
    int maxArea = 0;

    while (left < right) {
      var current = (right - left) * (Math.min(heights[left], heights[right]));
      if (maxArea < current) {
        maxArea = current;
      }

      if (heights[left] < heights[right]) {
        left++;
      } else {
        right--;
      }
    }
    return maxArea;
  }
}

File: Go/hellointerview/01-two-pointers/maxArea.go
package main

func MaxArea(heights []int) int {
	left, right := 0, len(heights)-1

	result := 0

	for left < right {
		currentArea := (left - right) * min(heights[left], heights[right])

		if currentArea > result {
			result = currentArea
		}

		if heights[left] < heights[right] {
			left++
		} else {
			right--
		}
	}

	return result
}
